jujugui/static/gui/src/app/views/topology/relation.js

```diff
@@ -1139,12 +1103,13 @@ YUI.add('juju-topology-relation', function(Y) {
       // Set up the relation data.
       var endpoint1 = endpoints[0][0],
           endpoint2 = endpoints[1][0];
-      var relation_id = 'pending-' + endpoint1 + endpoint2;
+      var idBlock = `${endpoint1}${endpoint2}`;
```

Comment:
these 3 line modifications was the actual fix, everything else was just dead code removal.